param/parameterized.py

```diff
@@ -2309,13 +2375,47 @@ def params(self_, parameter_name=None):
 
     def update(self_, arg=Undefined, /, **kwargs):
         """
-        For the given dictionary or iterable or set of param=value
-        keyword arguments, sets the corresponding parameter of this
-        object or class to the given value.
+        Update one or more parameters of this object or class.
+
+        Allows setting the parameters of the object or class using a dictionary, an iterable, or keyword arguments
+        in the form of `param=value`. The specified parameters will be updated to the given values.
+
+        This method can also be used as a context manager to temporarily set and then reset parameter values.
```

Comment:
Does `update` let you set a bunch of parameter values at once without triggering events until all of them have been set? I may be confused about some other method, but if this is what lets you do that, it should presumably be mentioned in the docstring.

Reply:
Yes, usually we recommend updating multiple parameters at once with `update` instead of sequentially, specially in a Panel app where this can lead to multiple re-renders.

Reply:
I'm not sure how best to express that in terms that make sense at the Param level, and didn't see much in the code that would let me state precisely what is deferred during the update.  @philippjfr, can we say that events are not delivered to watchers until all parameter values are updated?

Reply:
> Yes, usually we recommend updating multiple parameters at once with `update` instead of sequentially, specially in a Panel app where this can lead to multiple re-renders.

I'm always in doubt about `.update`. Does it batch updates. I've never seen that explained or demonstrated.

Reply:
I does update everything and then execute callbacks:

```python
import param

class MyParam(param.Parameterized):
    a = param.String()
    b = param.String()

my_param = MyParam(a="0. Hello", b="0. World")

@param.depends(my_param.param.a, watch=True)
def print_b(a):
    print(my_param.b)

my_param.a="1. Hello"
my_param.b="1. World"
my_param.param.update(a="2. Hello",b="2. World")
```

```bash
0. World
2. World
```

Reply:
I*ve updated the docstring based on the feedback.